config/routes.rb

```diff
@@ -2,6 +2,11 @@
   crud = %i(index show create update destroy)
 
   namespace :v1 do
-    resources :projects, only: crud
+    resources :projects, only: crud do
+      resources :tasks, only: [:create, :index]
+      # Redirect more RESTful nested requests to /projects/:project_id/tasks/:task_id/ to /tasks/:task_id
+      get '/tasks/:id', to: redirect('/v1/tasks/%{id}')
+    end
+    resources :tasks, only: [:show]
```

Comment:
When I had to decide how to setup the routes for this project, I tried to find out what the Rails convention was. What influenced the design the most was this part of the docs: http://guides.rubyonrails.org/routing.html#shallow-nesting
that says:
> One way to avoid deep nesting (as recommended above) is to generate the collection actions scoped under the parent, so as to get a sense of the hierarchy, but to not nest the member actions. In other words, to only build routes with the minimal amount of information to uniquely identify the resource

I tried to follow this by putting the two collection actions nested, and having `tasks#show` operate at the top level. I added the redirect to support the standard nesting path, while still only having 1 real `tasks#show` endpoint and URL. 

This could eventually be cleaner as just doing
`resources :projects do`
  resources :tasks, shallow: true
end`

like the docs suggest, but I kept it explicitly only the actions that were a part of this project for now.

I'm not really sure if this is the convention at all for routing and managing child endpoints, but I think it is one approach that is reasonable and seems to have some backing in the community.